doc/usecases/core/UC_keyset_lookup_hierarchy.md

```diff
@@ -1,23 +1,23 @@
-# Use Case: Cut `Key` hierarchy from `KeySet`
+# Use Case: Lookup `Key` hierarchy from `KeySet`
 
 ## Summary
 
 - **Scope:** `libelektra-core`
 - **Level:** Developer Goal
 - **Actors:** Core, Caller
-- **Brief:** Caller cuts (=looks up and removes) a `Key` hierarchy from an existing `KeySet`
+- **Brief:** Caller looks up a `Key` hierarchy from an existing `KeySet`
 
 ## Scenarios
 
 - **Precondition:**
   - [`KeySet` has been created](UC_keyset_create.md)
 - **Main success scenario:**
-  - Caller asks Core to cut a `Key` hierarchy descendant from a given `Key` R from a `KeySet` KS.
-  - Core searches for the first `Key` in KS that is part of the desired hierarchy, i.e., the first `Key` in KS that is a descendant of R according to [`Key` hierarchy comparison](UC_keyname_hierarchy.md).
-  - If not found, Core returns an empty KS
-  - If found, Core searches for the last `Key` that is not part of the desired hierarchy, i.e., Core tries all `Key`s in KS in [keyname order](UC_keyname_ordering.md) until it finds a `Key` that is not a descendant of R.
-  - Core removes all `Key`s of the desired hierarchy from KS and puts them into a new `KeySet` KS1
-  - Core returns KS1.
+  - Caller asks Core to lookup a `Key` hierarchy descendant from a given `Key` R from a `KeySet` KS.
+  - Core searches for the first `Key` in KS that is part of the desired hierarchy, i.e., the first `Key` F in KS that is a descendant of R according to [`Key` hierarchy comparison](UC_keyname_hierarchy.md).
+  - If not found, Core returns an out-of-range index range.
+  - If found, Core searches for the first `Key` L that is not part of the desired hierarchy.
+    This is equivalent to Core trying all `Key`s in KS after F in [keyname order](UC_keyname_ordering.md) until it finds a `Key` that is not a descendant of R.
+  - Core returns an index range `[f, l)` (start inclusive, end exclusive), where `f` is the index of `Key` F and `l` the index of L.
```

Comment:
Please describe why we get an index range if we only wanted the "hierarchy descendant".

Reply:
Like above, we want the "hierarchy, which is descendant ...".

Returning an index range is
1. Consistent with lookup, which also only returns an index.
2. A lot more flexible than returning a `KeySet *`. The `KeySet *` needs to be allocated and a whole array of pointers has to be created (*). That would be totally unnecessary, if all the caller wants to do is iterate over the `Key`s in the hierarchy, e.g. to print them. It would be even worse with just a `ksCut` and not a `ksBelow`. Then you have to reinsert the keys you wanted to iterate too.

(*) The array allocation part could be solved by adding an `size_t offset` field to `KeySet` and letting `KeySet`s share arrays with different offsets. But that makes things more complicated than they need to be.